### DISCUSSION TITLE
TypeError: Only absolute URLs are supported

### DISCUSSION THREAD

**Discussion by Jagadesh-Ram** [vercel/next.js#14492] [Help] [answered]
Am trying to use this,
`
export const getStaticProps = async (params) => {
  console.log("getStaticProps called");
  const data = await fetch(
    firebase
      .database()
      .ref("Memes/")
      .limitToLast(50)
      .orderByKey()
      .on("value", function (snapshot) {
        console.log(`snapshot ${JSON.stringify(snapshot)}`);
        resolve(snapshot.val());
      })
  );
  const res = await data.json().then((items) => items);
  console.log(res);
  return {
    props: {
      res,
    },
  };
};
`

**Comment by 0jg** [comment, marked as answer]
You need to use the full URL of your API endpoint. i.e., with the http and root of the URL: `http://<root-url>/api/...`

**Reply by Jagadesh-Ram** [reply]
yes man, i tried them before it worked but i just forgot here to update
